src/draw_target.rs

```diff
@@ -456,15 +461,60 @@ impl RateLimiter {
 
 const MAX_BURST: u8 = 20;
 
+#[derive(Debug, Clone, PartialEq)]
```

Comment:
We have a top-down order here, this should move down below any types that rely on it (`DrawState` at least).